File: searching-sorting-algo/01-searching/inJava/01-linear-search.java
import java.util.Scanner;

class Example{
    public static void main(String[] args){

        int[] ARRAY  = {10, 45, 5, 50, 2};
        int x = 2;
        linearSearch(ARRAY, x);
    }

    public static void linearSearch(int array[], int x){
        for(int i = 0; i<array.length; i++){
            if(x == array[i]){
                System.out.println("Element Found in Array");
                return;
            }
        }

        System.out.println("Element Not Found In Array");
    }
}

File: searching-sorting-algo/01-searching/01-linear-search.c
#include <stdio.h>
#include <stdlib.h>

// linear search or sequential search
/*
    searching technique is done in sequential manner
    TC = O(n)

    seqSearch(A, i, x, n)
    where, 
        A is array 
        i is index number
        x is element to search
        n is length of array
*/

int linearSearch(int array[], int n, int x){
    for(int i=0; i<n; i++)
        if(x == array[i])
            return 1;
        
    return -1;
}

int main(){

    int ARRAY[5] = {1, 3, 5, 12, 60};
    int x = 120;
    // length of array 
    int n = sizeof(ARRAY) / sizeof(ARRAY[0]);

    int result = linearSearch(ARRAY, n, x);

    (result == 1)? printf("Element Found"):printf("Element Not Found");

    return 0;
}